api/observability/v1/filter_types.go

```diff
@@ -0,0 +1,113 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import "github.com/openshift/cluster-logging-operator/internal/utils/sets"
+
+// Filter type constants, must match JSON tags of FilterTypeSpec fields.
+const (
+	FilterAddLabels       = "addLabels"
+	FilterDetectMultiline = "detectMultiline"
+	FilterDrop            = "drop"
+	FilterKubeAPIAudit    = "kubeAPIAudit"
+	FilterParse           = "parse"
+	FilterPrune           = "prune"
+)
+
+var ReservedFilterTypeNames = sets.NewString(FilterAddLabels, FilterDetectMultiline, FilterKubeAPIAudit, FilterDrop, FilterParse, FilterPrune)
+
+// FilterSpec defines a filter for log messages.
+type FilterSpec struct {
+	// Name used to refer to the filter from a `pipeline`.
+	//
+	// +kubebuilder:validation:Required
+	Name string `json:"name"`
+
+	// Type of filter.
+	//
+	// +kubebuilder:validation:Enum:=addLabels;detectMultiline;drop;kubeAPIAudit;parse;prune
+	// +kubebuilder:validation:Required
+	Type string `json:"type"`
```

Comment:
Seems like I forgot to change this to a separate type. Do we want to do this?

Reply:
Updating for consistency